miriway.cpp

```diff
@@ -94,9 +95,25 @@ struct ShellComponents
             launch(command);
         }
     }
+
+    bool try_add_lockscreen_app(std::vector<std::string> const& cmd)
+    {
+        if (lockscreen_app.has_value())
+            return false;
+
+        lockscreen_app = cmd;
+        return true;
+    }
+
+    void launch_lockscreen() const
+    {
+        if (lockscreen_app.has_value())
+            launch(lockscreen_app.value());
+    }
 private:
     std::function<void (std::vector<std::string> const& command_line)> const launch;
     std::vector<std::vector<std::string>> commands;
+    std::optional<std::vector<std::string>> lockscreen_app;
```

Comment:
There's only zero or one lockscreen_app

```suggestion
    std::optional<std::string> lockscreen_app;
```